app/components/course-page/course-stage-step/second-stage-instructions-card/implement-solution-step.hbs

```diff
@@ -15,4 +16,36 @@
       tab.
     {{/if}}
   </p>
-</div>
\ No newline at end of file
+</div>
+
+{{#if (and @shouldShowSolution this.solution)}}
+  <BlurredOverlay @isBlurred={{this.solutionIsBlurred}}>
+    <:content>
+      <div class="flex flex-col gap-4 my-3">
+        {{#each this.solution.changedFiles as |changedFile|}}
+          {{! Extra if condition convinces typescript that solution isn't null }}
+          {{#if this.solution}}
+            <FileDiffCard @filename={{changedFile.filename}} @code={{changedFile.diff}} @language={{this.solution.language.slug}} />
+          {{/if}}
+        {{/each}}
+      </div>
+
+      {{#if @shouldRecommendLanguageGuide}}
+        <p class="prose dark:prose-invert prose-compact">
+          For a more detailed explanation on how this solution works, view the
+          <a href="#language-guide-card">{{@repository.language.name}} Guide</a>
+          card.
+        </p>
+      {{/if}}
+    </:content>
+
+    <:overlay>
+      <PrimaryButton {{on "click" this.toggleSolution}} class="self-center">
```

Comment:
Another minor naming note: 

```suggestion
      <PrimaryButton {{on "click" this.handleRevealSolutionButtonClick}} class="self-center">
```

I know that the current implementation is consistent with autofix-result.hbs, so no mistake on your part - that file seems to be a mistake though, we usually don't name click handlers like this. There are two problems here: 

1. We don't actually support "toggling", we only support unblurring/revealing a solution. This unintentionally suggests that toggling is possible. This adds a bunch of useless code - like needing to write `this.solutionIsBlurred = !this.solutionIsBlurred` instead of the simpler `this.solutionIsBlurred = false`

2. Our click handlers are are usually named like `this.handleClick` / `this.handleXYZClick`. This reduces the naming decisions we have to make + allows us to add functionality to the click handler without having to change the name. Hypothetical  example, if here we decided to show a popup before toggling the solution, we'd have to both change the implementation and also change the name to something like `openRevealSolutionConfirmationModal`. If the name was `handleRevealSolutionButtonClick`, we'd only need to change the implementation of the function, the name would still be valid.